plugins/file/plugin.go

```diff
@@ -21,16 +26,78 @@ func init() {
 	plugins.RegisterPlugin("file", setupFile6, setupFile4)
 }
 
+//Record holds an IP lease record
+type Record struct {
+	IP        net.IP
+	leaseTime time.Duration
+	leased    int64
+}
+
 // StaticRecords holds a MAC -> IP address mapping
 var StaticRecords map[string]net.IP
 
 // DHCPv6Records and DHCPv4Records are mappings between MAC addresses in
 // form of a string, to network configurations.
 var (
 	DHCPv6Records map[string]net.IP
-	DHCPv4Records map[string]net.IP
+	DHCPv4Records map[string]Record
+	LeaseTime     time.Duration
+	filename      string
+	server        net.IPNet
 )
 
+// LoadDHCPv4Records loads the DHCPv4Records global map with records stored on
+// the specified file. The records have to be one per line, a mac address and an
+// IPv4 address.
+func LoadDHCPv4Records(filename string) (map[string]Record, error) {
+	log.Printf("plugins/file: reading leases from %s", filename)
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	records := make(map[string]Record)
+	for _, lineBytes := range bytes.Split(data, []byte{'\n'}) {
+		line := string(lineBytes)
+		if len(line) == 0 {
+			continue
+		}
+		tokens := strings.Fields(line)
+		if len(tokens) != 4 {
+			return nil, fmt.Errorf("plugins/file: malformed line: %s", line)
+		}
+		hwaddr, err := net.ParseMAC(tokens[0])
+		if err != nil {
+			return nil, fmt.Errorf("plugins/file: malformed hardware address: %s", tokens[0])
+		}
+		ipaddr := net.ParseIP(tokens[1])
+		if ipaddr.To4() == nil {
+			return nil, fmt.Errorf("plugins/file: expected an IPv4 address, got: %v", ipaddr)
+		}
+
+		leaseTime, err := time.ParseDuration(tokens[2])
+		if err != nil {
+			return nil, fmt.Errorf("plugins/file: expected an uint32, got: %v", ipaddr)
+		}
+		leased, err := strconv.ParseInt(tokens[3], 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("plugins/file: expected an uint32, got: %v", ipaddr)
+		}
+		//Only if the record is not expired.
+		println()
```

Comment:
Are these two `println` going to be removed?

Reply:
Yes, my bad.